### ISSUE TITLE
Add support for `functools.total_ordering`

### ISSUE THREAD

**Issue by JimDabell** [astral-sh/ty#1202]
### Summary

Classes can define `__eq__()` and one of `__lt__()`, `__le__()`, `__gt__()`, or `__ge__()`, then use the `@functools.total_ordering` decorator to fill in the rest of the comparison dunder methods.

ty does not see these implementations and incorrectly reports `unsupported-operator`.

```python
import functools


@functools.total_ordering
class Comparable:
	def __init__(self, value: int):
		self.value = value

	def __eq__(self, other) -> bool:
		return self.value == other.value

	def __lt__(self, other) -> bool:
		return self.value < other.value

	# Uncommenting this implementation satisfies ty
	# def __le__(self, other) -> bool:
	# 	return self.value <= other.value


a = Comparable(1)
b = Comparable(2)

print(a <= b)
```

This may be related to #391.

Neither Pyright nor Mypy flag this as an error.

### Version

ty 0.0.1-alpha.20 (f41f00af1 2025-09-03)

**Comment by AlexWaygood** [comment]
Thanks for the report!

Mypy and pyright both apply special handling for this decorator, which monkey-patches methods onto a class in a similar way to the `@dataclass` decorator. We'll need to apply special handling too in order to support the decorator.
